components/managedidentity/04-managed-identities.tf

```diff
@@ -51,6 +51,21 @@ resource "azurerm_cosmosdb_sql_role_assignment" "identity_contributor" {
   scope              = data.azurerm_cosmosdb_account.version_reporter.id
 }
 
+data "azurerm_cosmosdb_account" "pipeline_metrics" {
+  provider            = azurerm.pipeline-metrics
+  name                = local.cosmosdb_name
+  resource_group_name = local.cosmosdb_rg
+}
+
+resource "azurerm_cosmosdb_sql_role_assignment" "monitoring_mi_assignment" {
+  provider            = azurerm.ptl
```

Comment:
should this provider match the data lookup; `provider            = azurerm.pipeline-metrics`